### DISCUSSION TITLE
Securing subscription feeds

### DISCUSSION THREAD

**Discussion by JipFr** [supabase/supabase#3714] [Questions] [answered]
Hey all.

I'm converting my app from a small self-hosted API to Supabase. I have the following security policies, but when a subscription is made like the following piece of code, all connected clients will receive all events. Is there any way to stop this from happening, as the data is intended to be private?

```js
SupabaseClient.from('transactions')
      .on('*', (payload) => {
            // Do stuff here
      })
      .subscribe()
```

![Screenshot 2021-10-30 at 00 56 56](https://user-images.githubusercontent.com/17685409/139509690-08e70f17-f286-48f0-ba45-2cd4af47c1b2.png)

Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
Right now Realtime ignores all RLS settings.  I have an event table that I subscribe to that is separate from my real data that gets updated with triggers so the client only sees flags/ids.  Then I fetch data as appropriate from the RLS secured table.  Supabase is working on realtime with RLS called Walrus, but no ETA has been posted that I've seen.

**Comment by JipFr** [comment]
Thanks for the swift response! Is there any place I can follow the development of Walrus?

**Reply by GaryAustin1** [reply]
supabase/walrus github

**Reply by JipFr** [reply]
Thank you!

**Comment by w3b6x9** [comment]
@JipFr just wanted to give you an update that all Supabase projects now have Realtime RLS, which means Realtime will only broadcast to authorized users based on your in place RLS policies.

**Reply by JipFr** [reply]
Awesome, thanks!

**Reply by JipFr** [reply]
I know this is very late, but I just took the code I made back then and applied it to my current code base, and it seems to work flawlessly! I checked with another account and they didn't seem to send each other events. Thanks again!
